custom/dhis2/models.py

```diff
@@ -0,0 +1,578 @@
+# from django.db import models
+from datetime import date
+from corehq.apps.fixtures.models import FixtureDataItem, FixtureDataType, FieldList, FixtureItemField
+from django.conf import settings
+import requests
+
+
+# TODO: Move to init
+DOMAIN = 'wv-lanka'
+LOOKUP_TABLE = 'dhis2_org_unit'
+
+
+class JsonApiError(Exception):
+    """
+    JsonApiError is raised for HTTP or socket errors.
+    """
+    pass
+
+
+class Dhis2ApiQueryError(JsonApiError):
+    """
+    Dhis2ApiQueryError is raised when the API returns an unexpected response.
+    """
+    pass
+
+
+class Dhis2ConfigurationError(Exception):
+    pass
+
+
+class JsonApiRequest(object):
+    """
+    Wrap requests with URL, header and authentication for DHIS2 API
+    """
+
+    def __init__(self, host, username, password):
+        self.baseurl = host + '/api/'
+        self.headers = {'Accept': 'application/json'}
+        self.auth = (username, password)
+
+    @staticmethod
+    def json_or_error(response):
+        """
+        Return HTTP status, JSON
+
+        :raises JsonApiError if HTTP status is not in the 200 (OK) range
+        """
+        if 200 <= response.status_code < 300:
+            return response.status_code, response.json()
```

Comment:
Other than in tests, is `response.status_code` ever used? I'm seeing that most places it's thrown away. Wondering if it's necessary?